sympy/functions/elementary/hyperbolic.py

```diff
@@ -189,12 +189,22 @@ def _eval_as_leading_term(self, x):
             return self.func(arg)
 
     def _eval_is_real(self):
-        return self.args[0].is_real
+        from sympy import pi
+        arg = self.args[0]
+        if arg.is_real:
+            return True
+
+        # if `im` is of the form n*pi
+        # else, check if it is a number
+        re, im = arg.as_real_imag()
+        if (im % pi) == 0:
+            return True
+        elif im.is_number:
```

Comment:
I don't understand - why do we return `False` if imaginary part of argument `is_number`. Can you explain it, I just don't know why it's decisive.